### DISCUSSION TITLE
Tool for endpoints testing with auto updates on client side

### DISCUSSION THREAD

**Discussion by dufguix** [serverpod/serverpod#1955] [Q&A] [answered]
Hello

I'm trying to make a flutter app to test the server endpoints. in the same idea of Postman, Swagger, ...
I'd like to avoid updating the ui after server updates.

Any idea to get methods name and parameters during runtime ?
The only thing I found is 'client.endpointRefLookup' for endpoint names.

Or maybe during code generation ?
I have no experience with code generation in dart but can give it a try.

Or maybe there is an existing tool ?

**Comment by SandPod** [comment, marked as answer]
Hello @dufguix,

As far as I know, there currently doesn't exists a dynamic way to fetch all the endpoints.
We do however generate a file called `protocol.yaml` in  `<your_project_name>_server/lib/src/generated` that lists your defined endpoints with their methods.

**Reply by dufguix** [reply]
Thanks for your reply.
